### PR TITLE
Update UG and DG implementation for `attendance` `mark` and `unmark` command

### LINKED ISSUE #148: Update UG and partial DG for `attendance` `mark` and `unmark` command

### LINKED ISSUE #168: Update implementation of `mark` and `unmark` in DG

### PR DESCRIPTION
**Pull request by Rachelcoll**
- Fixes #148 
- Fixes #168

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -423,8 +426,70 @@ Use Case ends.
...
+**MSS**
+
+1. User requests to mark the attendance of members.
+2. User provides the details of the session (date) to mark attendance, and list of telegrams of members to be marked.
+3. System marks the attendance of the list of members on the specified date.
+
+Use case ends.
+
+**Extensions**
+* 8a. System detects that user did not enter all the required fields.
```
Comment (weiliann) [inline docs/DeveloperGuide.md]: I think the numbering should match the step numbers. so 1a in this case

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -423,8 +426,70 @@ Use Case ends.
...
+    * 8a1. System shows user an error message.
+    * Use case ends.
```
Comment (weiliann) [inline docs/DeveloperGuide.md:421]: i think the use case doesnt end when the input is invalid? i think either it repeats or have another step where the user re-enters the details

File: docs/UserGuide.md [hunk 3]
```diff
@@ -141,6 +141,70 @@ Examples:
...
+  * Telegram handle should belong to a contact with `Member` role.
+
+- Date (`d/date`)
+  * Takes in one string value with format d/YYYY-MM-DD (eg. d/2024-01-22 not d/2024/1/22)
+  * Only accept one date, if multiple dates are input, only the last one will be recorded as the attendance date.
+  * Date must not be later than current date (eg. 2090-11-02 is an invalid datetime)
+
+Examples:
+
+* `mark t/toom t/maary d/2024-11-02`
```
Comment (weiliann) [inline docs/UserGuide.md:183]: is 'toom' and 'maary' intentional or typo haha

**Review by weiliann — CHANGES_REQUESTED** [review]
need to update the numbering for the extensions, and when the use case ends

File: docs/UserGuide.md (hunk 3 above)
Comment (Rachelcoll) [reply docs/UserGuide.md:183]: it's the telegram handle of the contacts, setting to distinguish from the contacts' actual name

Comment (weiliann) [reply docs/UserGuide.md:183]: oh ok

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Rachelcoll) [reply docs/DeveloperGuide.md:421]: I personally feels that in this case, if user input invalid commands, the state of the system will return to original state where user can input any other command (can be seen as another independent use case) if they don't want to reenter the detail for the same command again. Also other UCs in the given examples generally ends the usecase if user input incorrect message in the first step.

Comment (weiliann) [reply docs/DeveloperGuide.md:421]: ok

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Rachelcoll) [reply docs/DeveloperGuide.md]: Thanks, already updated.

**Review by weiliann — DISMISSED** [review]

**Review by weiliann — CHANGES_REQUESTED** [review]
Hi, I run gradle check locally but it doesn't show any checkstyle error in the DeveloperGuide.md file, I'm not so sure why it fails in github, do you know why?

**Comment by weiliann** [comment]
> Hi, I run gradle check locally but it doesn't show any checkstyle error in the DeveloperGuide.md file, I'm not so sure why it fails in github, do you know why?

The checkstyle is failing for the ubuntu os but works for windows. If you click on 'Details' beside the failed build, it shows the lines with additional white space at the end of the line that is causing the build to fail.

**Comment by Rachelcoll** [comment]
> > Hi, I run gradle check locally but it doesn't show any checkstyle error in the DeveloperGuide.md file, I'm not so sure why it fails in github, do you know why?
> 
> The checkstyle is failing for the ubuntu os but works for windows. If you click on 'Details' beside the failed build, it shows the lines with additional white space at the end of the line that is causing the build to fail.

Thanks, already updated

**Review by weiliann — CHANGES_REQUESTED** [review]
docs/images/ListAttendanceResult.png

i think the command is misspelled here

**Comment by Rachelcoll** [comment]
> docs/images/ListAttendanceResult.png
> 
> i think the command is misspelled here

Thanks for the review, already updated the image

**Review by weiliann — APPROVED** [review]

**Review by shizy — APPROVED** [review]

**Pull request merged**